hbase-zookeeper/src/main/java/org/apache/hadoop/hbase/zookeeper/ZKMainServer.java

```diff
@@ -41,7 +42,8 @@ public String parse(final Configuration c) {
    * ZooKeeper 3.4.6 broke being able to pass commands on command line. See ZOOKEEPER-1897. This
    * class is a hack to restore this faclity.
    */
-  private static class HACK_UNTIL_ZOOKEEPER_1897_ZooKeeperMain extends ZooKeeperMain {
+  private static class HACK_UNTIL_ZOOKEEPER_1897_ZooKeeperMain extends ZooKeeperMain
+      implements Closeable {
```

Comment:
Probably not scope of this ticket, but [ZOOKEEPER-1897](https://issues.apache.org/jira/browse/ZOOKEEPER-1897) resolved long ago. Do we still need this class?

Reply:
We have bumped our zookeeper dependency to 3.8.x, do we still need this?

Reply:
Thank you for reviewing the code and thank you for mentioning ZOOKEEPER-4804.
I did not know that non-daemon Netty threads issue had been resolved in ZooKeeper.

It seems that the `HACK_UNTIL_ZOOKEEPER_1897_ZooKeeperMain` class is no longer necessary.
Instead of using `HACK_UNTIL_ZOOKEEPER_1897_ZooKeeperMain`, it might be better to use `ZooKeeperMain` directly.
`ZooKeeperMain` also closes `ZooKeeper` when a timeout occurs.

However, it seems that the ZOOKEEPER-4804 patch is not applied in the 3.8.4 version used by HBase.
Therefore, in order to remove `HACK_UNTIL_ZOOKEEPER_1897_ZooKeeperMain`, it appears that a ZooKeeper version upgrade(3.8.4 -> 3.9.x) is necessary.

Two options are suggested below. Which option would be the most appropriate?

1. Upgrade ZooKeeper to version 3.9.x and remove `HACK_UNTIL_ZOOKEEPER_1897_ZooKeeperMain` (Personally prefer.)
Is it safe to upgrade ZooKeeper to version 3.9.x? I am concerned that other issues may arise.
2. Always ensure `HACK_UNTIL_ZOOKEEPER_1897_ZooKeeperMain` performs closing. (Current approach)

Reply:
Better also add ZOOKEEPER-4804 in the comment so others will know why we still need this hack.

Reply:
Thank you for your advice.
I have added ZOOKEEPER-4804 to the comments.

Reply:
Sounds good to me, thanks.

Reply:
I'm pleased to inform that the code review has been approved and the CI checks have passed successfully.
If possible, would you kindly merge this code? I apologize for any inconvenience, but I don't have the necessary permissions to perform the merge myself.
Alternatively, if there are any additional steps required to proceed with the merge, please advise me.